presto-main/src/main/java/com/facebook/presto/execution/DropColumnTask.java

```diff
@@ -50,23 +48,25 @@ public String getName()
     {
         Session session = stateMachine.getSession();
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getTable());
-        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+        TableHandle tableHandle = metadata.getTableHandle(session, tableName)
+                .orElseThrow(() -> new SemanticException(MISSING_TABLE, statement, "Table '%s' does not exist", tableName));
 
         String column = statement.getColumn().getValue().toLowerCase(ENGLISH);
 
-        if (!tableHandle.isPresent()) {
-            throw new SemanticException(MISSING_TABLE, statement, "Table '%s' does not exist", tableName);
-        }
         accessControl.checkCanDropColumn(session.getRequiredTransactionId(), session.getIdentity(), tableName);
 
-        Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle.get());
-        if (!columnHandles.containsKey(column)) {
+        ColumnHandle columnHandle = metadata.getColumnHandles(session, tableHandle).get(column);
+        if (columnHandle == null) {
             throw new SemanticException(MISSING_COLUMN, statement, "Column '%s' does not exist", column);
         }
-        if (columnHandles.size() == 1) {
-            throw new SemanticException(NOT_SUPPORTED, statement, "Cannot drop column from a table with only one column");
+        if (metadata.getColumnMetadata(session, tableHandle, columnHandle).isHidden()) {
+            throw new SemanticException(NOT_SUPPORTED, statement, "Cannot drop hidden column");
+        }
+        if (metadata.getTableMetadata(session, tableHandle).getColumns()
+                .stream().filter(info -> !info.isHidden()).count() <= 1) {
```

Comment:
the condition is quite long, consider pulling the `...count()` before the `if` (or to a method) for readability.

Reply:
I played around with different variants and moving the `.stream()` to the previous line along with blank lines around the if statement seemed most readable (to me).

Updated PR.